tasks/src/feature-extraction/about.md

```diff
@@ -1,15 +1,93 @@
+
+```python
+dataset =  datasets.fetch_20newsgroups()
+X = dataset.data[:100]
+y = dataset.target[:100]
+```
+This function creates a sparse embedding for each token present in the sentence. Please remember to use the same tokenizer that you would be using while calling the feature-extraction pipeline. As the default BERT tokenizer used in the pipeline is bert-base-cased so we use the same for initializing the dataframe with number of columns = vocabulary size of tokenizer. We are using dataframe within the function for ease of understanding and it returns an np.array of size (n_samples,vocab_size)
+
+```python
+def get_features(X,feature_extraction):
+
+    # Initializing the input dataframe
+    tokenizer = AutoTokenizer.from_pretrained('bert-base-cased')
+    df = pd.DataFrame(columns = sorted(tokenizer.vocab.values()))
+
+    # Looping over the input data
+    for i in tqdm(X):
+        tokenized_input = tokenizer(i)
+
+        # Getting embeddings for each token
+        features = feature_extraction(input_)
+        features = np.squeeze(features,axis = 0)
+
+        # Calculating mean of embeddings for each token over 768 embedding values
+        features = np.mean(features,axis = 1)
+
+        # Mapping token embeddings to their specific column in the data-frame
+        features_dict = dict(zip(tokenized_input['input_ids'],features))
+        df = df.append(features_dict, ignore_index=True)
+
+    # Filling NaN values with 0
+    df = df.fillna(0)
+    return df.values
+```
+Below we perform a 80:20 train-test split for modeling
+
+```python
+X_features = get_features(X,feature_extraction)
+X_train, X_test, y_train, y_test = train_test_split(X_features, y,stratify = y, test_size=0.2, random_state=42)
+```
+
+Performing Multi-class classification using sklearn Random Forest
+
+```python
+model = RandomForestClassifier(random_state=42)
+model.fit(X_train,y_train)
+y_pred = model.predict(X_test)
+print(f'Test Accuracy is {accuracy_score(y_test,y_pred)*100} %')
+
+# Output:
+# Test Accuracy is 95.0 %
+```
+This exercise shows the potential of using embeddings as an input to a classifier. For imbalanced dataset, we can use performance metrics like F1 and ROC_AUC.
+
+
+# Task Variants
 
 ## Inference
 
-You can infer with text data with the 🤗 Transformers library using the `feature-extraction` pipeline. This pipeline takes a string or a list of string input and generates features for each input. All models may be used for this pipeline (see a list of all models on https://huggingface.co/models).
+This pipeline extracts the hidden states from the base transformer, which can be used as features in downstream tasks. You can infer the text data with the 🤗 Transformers library using the `feature-extraction` pipeline. This pipeline takes a string or a list of string input and generates features for each input. All models may be used for this pipeline (see a list of all models on https://huggingface.co/models).
```

Comment:
Can you try to explain what hidden state is at the beginning? Try to read the text as a beginner who doesn't know much about the task or ML 🙂

Reply:
Can you please address this comment? :')